Sources/performance-worker/BenchmarkService.swift

```diff
@@ -0,0 +1,135 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+
+import struct Foundation.Data
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+struct BenchmarkService: Grpc_Testing_BenchmarkService.ServiceProtocol {
+  /// One request followed by one response.
+  /// The server returns the client payload as-is.
+  func unaryCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_BenchmarkService.Method.UnaryCall.Input>
+  ) async throws
+    -> GRPCCore.ServerResponse.Single<Grpc_Testing_BenchmarkService.Method.UnaryCall.Output>
+  {
+
+    // Throw an error if the status is not `ok`. Otherwise, an `ok` status is automatically sent
+    // if the request is successful.
+    if request.message.responseStatus.isInitialized {
+      try self.echoStatus(responseStatus: request.message.responseStatus)
+    }
+
+    return ServerResponse.Single(
+      message: Grpc_Testing_BenchmarkService.Method.UnaryCall.Output.with {
+        $0.payload = request.message.payload
+      }
+    )
+  }
+
+  /// Repeated sequence of one request followed by one response.
+  /// The server returns the client payload as-is on each response
+  func streamingCall(
+    request: GRPCCore.ServerRequest.Stream<Grpc_Testing_BenchmarkService.Method.StreamingCall.Input>
+  ) async throws
+    -> GRPCCore.ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingCall.Output>
+  {
+    return ServerResponse.Stream { writer in
+      for try await message in request.messages {
+        if message.responseStatus.isInitialized {
+          try self.echoStatus(responseStatus: message.responseStatus)
+        }
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = message.payload
+          }
+        )
+      }
+      return [:]
+    }
+  }
+
+  /// Single-sided unbounded streaming from client to server
+  /// The server returns the client payload as-is once the client does WritesDone
+  func streamingFromClient(
+    request: ServerRequest.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Input>
+  ) async throws
+    -> ServerResponse.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Output>
+  {
+    throw RPCError(code: .unimplemented, message: "The RPC is not implemented.")
+  }
+
+  /// Single-sided unbounded streaming from server to client
+  /// The server repeatedly returns the client payload as-is
+  func streamingFromServer(
+    request: ServerRequest.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Input>
+  ) async throws
+    -> ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Output>
+  {
+    return ServerResponse.Stream { writer in
+      if request.message.responseStatus.isInitialized {
+        try self.echoStatus(responseStatus: request.message.responseStatus)
+      }
+
+      while true {
```

Comment:
We need to stop eventually, `while !Task.isCancelled` is probably a better condition to use here.